core/templates/dev/head/player/PlayerServices.js

```diff
@@ -315,6 +317,7 @@ oppia.factory('oppiaPlayerService', [
             }
           }
 
+          $location.path('/preview/' + newStateName);
```

Comment:
If I don't do this, navigating through the preview doesn't update the path, so clicking edit after the navigation won't stay on the same state.  I've wrapped it with `if (_editorPreviewMode)` to prevent it affecting the learner view.  Is that ok?